### ISSUE TITLE
[FR]: Lock position of article previewer window (verticle layout)

### ISSUE THREAD

**Issue by RetroAbstract** [martinrotter/rssguard#766]
### Brief description of the feature request

This feature would allow the user to lock the position of the article previewer window. 

Reason: The positioning of the article previewer gets automatically adjusted when moving RSS Guard from one monitor to another with a different resolution, resulting in having to always re-adjust it's positioning.

Would be located in: Right click in article previewer window > Lock window position.

**Comment by martinrotter** [comment]
You mean position of the "separator" between article previewer and article list? Can you perhaps make a video of the problem?

**Comment by akinokonomi** [comment]
> You mean position of the "separator" between article previewer and article list?

@martinrotter, That's how I understand it too.

Here: 

> article previewer gets automatically adjusted when moving RSS Guard from one monitor to another with a different resolution

Most likely cause is that RSS Guard sets article previewer size in pixels, not proportional to the total window size (e.g. in percents).

If you can confirm it then that where the problem is.

I can't test it myself unfortunately, so it's just a speculation.

> Would be located in: Right click in article previewer window > Lock window position.

I don't think it needs to have a separate option, it should work properly by default.

That is, unless I misunderstand something, and this is a duplicate of https://github.com/martinrotter/rssguard/issues/424, about persistent article previewer.

**Comment by martinrotter** [comment]
1. Yes, value for "splitter" between article list and article viewer is set in pixels.
2. I do not really do anything manual when window is moved between screens, all recalculations are done in upstream Qt libraries.

FOR ME: https://lists.qt-project.org/pipermail/interest/2016-October/024823.html

**Comment by akinokonomi** [comment]
> all recalculations are done in upstream Qt libraries.

@martinrotter Indeed, it can be the upstream issue, and its integration of QtWebengine into UI.
Qt6 supposed to have fixed their UI issues with screen scaling since Qt5, though. (No idea about Webengine.)

So you are saying pixels is the only way to keep coordinates of the article browser splitter? 
Does the feed list has it the same?

To clarify it for me, you just keep the position of splitter, not the size of the article viewer? 

---

> previewer gets automatically adjusted when moving RSS Guard from one monitor to another with a different resolution

@RetroAbstract, Do you experience it with nowebengine version of RSS Guard too? Could you test it please.

Also, just to confirm, do you have different scaling on those screens? And, you are on Windows, right?

**Comment by akinokonomi** [comment]
>  All recalculations of ratios etc. when transferring window to another screen are somehow done automatically.

Notice that OP's having problems only with article viewer and not with feed list.

**Comment by RetroAbstract** [comment]
> @RetroAbstract, Do you experience it with nowebengine version of RSS Guard too? Could you test it please.

I use the nowebengine version by default.


> Also, just to confirm, do you have different scaling on those screens? And, you are on Windows, right?

Yes I have different scaling percentages & resolutions on the 2 monitors, and yes I'm on Windows.

**Comment by martinrotter** [comment]
@RetroAbstract Can you re-test this with latest Qt6-based 4,.2.4 build?

**Comment by RetroAbstract** [comment]
@martinrotter Update: It seems RSS Guard will only remember the article previewer window position up until the system is turned off. It will then be reset to default the next time the system's turned back on.

Would be curious to know if this happens to others or if this is an isolated case.

**Comment by martinrotter** [comment]
So the problem is now de-facto solved?

**Comment by xfzv** [comment]
> but rather when RSS Guard is launched after being closed for a while

That's what I'm experiencing too, even with 4.2.4. The preview pane is completely shrunk and I have to manually resize it every time. Occurs with both vertical and horizontal view.

**Comment by RetroAbstract** [comment]
> What's the status of this? just curious.

Same as when I opened this issue

**Comment by andrewpros** [comment]
Well i planned to report this as a BUG as even on single screen the vertical article pane separator gets reset now and there, but im not sure its is still an issue with the newest version, need to check.

**Comment by andrewpros** [comment]
Well i verified it and it is still the case with win 4.2.5 828697bd-nowebengine.

For me, i just need to exit, either by closing Windows or just closing the app by Quit option, its not every time so i dont know the trigger factor, but it happens after a max of couple of tries. After win restart its almost always.

**Comment by RetroAbstract** [comment]
> Well i verified it and it is still the case with win 4.2.5 [828697b](https://github.com/martinrotter/rssguard/commit/828697bd413bea44d2b6653e623279d4d92f6bb5)-nowebengine.
> 
> For me, i just need to exit, either by closing Windows or just closing the app by Quit option, its not every time so i dont know the trigger factor, but it happens after a max of couple of tries. After win restart its almost always.

Same experience as Andrew. Difficult to reproduce at will as even when Windows is restarted it might not happen.

**Comment by andrewpros** [comment]
Just close the app from the tray, it will very quickly reset.

Splitter name is m_messageSplitter
Saved only when changing app window size or moving the splitter
In my case, it resets splitter_messages_horizontal config.ini value to 100, 0

It happens after the app is loaded so its not saving it with wrong value, i dont even need to click any feed, just restart, windows shows, boom, splitter_messages_horizontal is reset, u dont even see the splitter yet

If i had to guess, thats why, u dont see it so it "does not exists" but the save event is triggered for some reason which is a common issue with resize events, but if it does not exits it put a default size there...

**Comment by andrewpros** [comment]
The thing is, it could be not the reboot, its what happened already when u run the app that day, and if it is what i suspect, the resize event, its qt based and system independent issue, normally u put some safe checks when it is needed for situations like this.

**Comment by Neules** [comment]
Using  4.3.1 nowebengine in Windows and the "bug" persists. Splitter gets down when I click on an article (the first you click, obviously, when you get opened RSS guard).

![image](https://user-images.githubusercontent.com/34288954/215040672-06f11fb9-4c5b-4e78-8427-98b48e4dba38.png)

It's happening the exact same behaviour in kubuntu 22.10 which is 4.0.x I think, and it comes with the webengine by default.

**Comment by andrewpros** [comment]
It will persist as i dont think it was fixed.

**Comment by martinrotter** [comment]
Still same behavior? I gave it some testing and I am just not able to reproduce. Feel free to re-open or more preferably provide PR fis for this. I am just not able to reproduce myself on two machines (Windows 10 and Arch).
